### PR TITLE
feat:base style update "rebased"

### PR DESCRIPTION
**Pull request by ZenVega**

### CODE REVIEW

File: app/src/layout.py [hunk 1]
```diff
@@ -23,11 +24,25 @@
...
-    flex_obj = FlexBox(*plotter.plots.values(),
+    flex_obj = FlexBox(location_toggle,
+                       #    *plotter.plots.values(),
+                       plotter.plots[location_toggle.value],
                        *plotter.filters.values(),
                        flex_direction="column",
                        align_items="center",
                        sizing_mode="stretch_width")
 
+    update_chart = create_update_chart(flex_obj, plotter)
```
Comment (guadiromero) [inline app/src/layout.py:45]: I think we can simplify the code a lot by deleting this line and replacing the `create_update_chart` function with:

```
def update_chart(self, event):
    flex_obj.__setitem__(1, plotter.plots[event.new])
```

(I didn't test it, lmk if it doesn't work and we can figure it out together)

File: app/src/plotter.py [hunk 2]
```diff
@@ -108,10 +113,12 @@ def fit_data(self):
                 for choice_2 in self.config["filters"]["single_choice_2"]:
                     x_values = self.raw_data[code][choice][choice_2]["x"]
                     # Define color palette
-                    self.raw_data[code][choice][choice_2]["color"] = Category20[len(x_values)]
+                    self.raw_data[code][choice][choice_2]["color"] = Category20[len(
+                        x_values)]
                     # Split long x-axis in more than one line
                     for i, label in enumerate(x_values):
-                        self.raw_data[code][choice][choice_2]["x"][i] = "/\n".join(label.split("/"))
+                        self.raw_data[code][choice][choice_2]["x"][i] = "/\n".join(
```
Comment (guadiromero) [inline app/src/plotter.py:120]: This reformatting is not necessary

File: app/src/plotter.py [hunk 3]
```diff
@@ -123,19 +130,21 @@ def fit_data(self):
     def create_plot(self):
         for code in self.config["plot_codes"]:
             # Create the figure
-            plot = figure(x_range=self.fitted_data[code].data["x"], **self.config["general"])
+            plot = figure(
```
Comment (guadiromero) [inline app/src/plotter.py:133]: This reformatting is not necessary

File: app/src/plotter.py [hunk 4]
```diff
@@ -153,7 +162,8 @@ def create_filters(self):
 
         # Add interactivity
         self.filters["single_choice"].param.watch(self.update_filters, "value")
-        self.filters["single_choice_2"].param.watch(self.update_filters, "value")
+        self.filters["single_choice_2"].param.watch(
```
Comment (guadiromero) [inline app/src/plotter.py:165]: This reformatting is not necessary

File: app/src/plotter.py [hunk 5]
```diff
@@ -195,8 +206,10 @@ def create_plot(self):
                           y_range=[min(y_range), max(y_range)])
 
             # Add circles to the plot
-            mapper = linear_cmap(field_name="color", palette=Category20[20], low=0, high=20)
-            plot.circle(x="x", y="y", size="z", color=mapper, source=self.fitted_data[code], legend_group="labels")
+            mapper = linear_cmap(field_name="color",
```
Comment (guadiromero) [inline app/src/plotter.py:209]: This reformatting is not necessary

File: app/src/plotter.py [hunk 6]
```diff
@@ -257,52 +271,91 @@ def fit_data(self):
...
+            # Change background color
+            self.plots[code].background_fill_color = self.config["background_fill_color"]
+            self.plots[code].border_fill_color = self.config["background_fill_color"]
+
+            # Hide x grid
+            self.plots[code].xgrid.grid_line_color = None
+
+            # Configure axes
             self.plots[code].xaxis.ticker = x_range
-            self.plots[code].yaxis.formatter.use_scientific = False
```
Comment (guadiromero) [inline app/src/plotter.py:276]: Let's not delete this line

File: app/src/plotter.py [hunk 7]
```diff
@@ -15,6 +16,10 @@
     "pie_interactive": "InteractivePiePlotter",
 }
 
+overwrite_colors = [
```
[the remark this answers is not in the document]
Comment (guadiromero) [reply app/src/plotter.py:19]: I think we can also give it another name, like "custom_palette" or something like this

File: app/src/plotter.py [hunk 8]
```diff
@@ -257,52 +271,91 @@ def fit_data(self):
...
             # Add lines to the plot
-            colors = Category20[20]
+            colors = overwrite_colors
             for i, line_name in enumerate(self.config["filters"]["multi_choice"]):
-                self.plots[code].line(x="x", y=line_name, source=self.fitted_data[code], color=colors[i], legend_label=line_name)
+                self.plots[code].line(
+                    x="x", y=line_name, source=self.fitted_data[code], color=colors[i], line_width=4)
+                self.plots[code].circle(
+                    x="x", y=line_name, source=self.fitted_data[code], color=colors[i], size=16)
+                self.plots[code].circle(
```
Comment (guadiromero) [inline app/src/plotter.py:320]: Does this line overwrite the previous one? If so, we can delete the previous one.

File: app/src/plotter.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (ZenVega) [reply app/src/plotter.py:19]: Agreed!

File: app/src/layout.py (hunk 1 above)
Comment (ZenVega) [reply app/src/layout.py:45]: Hmm. I didn't get it running and from how I understand it, the factory function is needed, because we have multiple flexboxes to update. but we can look into it again.

File: app/src/plotter.py [hunk 9]
```diff
@@ -257,52 +271,91 @@ def fit_data(self):
...
+            panel.Row(
+                panel.pane.HTML(
+                    '<div class="legend-field" style="background-color:{};"></div>'.format(color)),
+                panel.widgets.Checkbox(name=option, value=True,
+                                       css_classes=["legend-checkbox"])
+            )
+            for color, option in zip(overwrite_colors, self.config["filters"]["multi_choice"])
+        ])
+
+        self.filters["multi_choice"] = Accordion(
```
Comment (guadiromero) [inline app/src/plotter.py:342]: As you use the same `header_color`, etc. for both filters, you can define these variables once before this line

File: app/src/plotter.py [hunk 10]
```diff
@@ -257,52 +271,91 @@ def fit_data(self):
...
+            ("Branchen auswählen", filters_multi_choice), header_color="#1E3791", active_header_background="#F6F6F6", header_background="#F6F6F6", active=[0])
+        self.filters["single_choice"] = Accordion(
+            ("Einheiten auswählen", filters_single_choice), header_color="#1E3791", active_header_background="#F6F6F6", header_background="#F6F6F6")
 
         # Add interactivity
-        self.filters["multi_choice"].param.watch(self.update_filters, "value")
-        self.filters["single_choice"].param.watch(self.update_filters, "value")
-        self.filters["single_choice"].param.watch(self.update_y_range, "value")
+        for filter_row in self.filters["multi_choice"][0]:
```
Comment (guadiromero) [inline app/src/plotter.py:348]: I'm having a hard time understanding the structure of this filter and how many rows and columns there are inside. Let's have a call later and draw it together to clarify the logic :) I feel like we don't need to use loops here.

File: app/styles/main.css [hunk 11]
```diff
@@ -58,6 +58,87 @@ body {
   @apply !bg-white !text-corporate-blue;
 }
 
+.bk-panel-models-widgets-CustomSelect {
```
Comment (guadiromero) [inline app/styles/main.css:64]: @herrherrmann 

Could you maybe review the HTML / CSS / JS code in this PR? 🌝
I think you can do it faster and give more meaningful feedback than me

File: app/src/plotter.py (hunk 10 above)
Comment (ZenVega) [reply app/src/plotter.py:348]: Sure thing! So `self.filters["multi_choice"]` holds an Accordion component now which is basically a set. Here is a log of it `logging.info(f"multi_choice: {type({self.filters['multi_choice']})}, {self.filters['multi_choice']}")`
image

so by looping over index 0, I iterate over every filter row, that consists of a div ( [0] HTML(str) ) and a Checkbox ( [1] Checkbox(css_classes=['legend-checkbox'], name='nahrung', value=True) ). 
So if the "value" of a checkbox is True, it is appended to the selected_lines: `if filter_row[1].value:
                    selected_lines.append(filter_row[1].name)`

Something equal happened to the single_choice_filters. Since we moved the complete rendering to the plotter.py file, `self.filters['single_choice']` now holds the Accordion and the actual RadioBoxGroup went one hierarchy lower:
`logging.info(f"single_choice: {self.filters['single_choice']}")` results in:
image
so to actually read out the value of the multi_choice_filter, like "anzahl_unternehmen_insges" we have to look for `self.filters['single_choice'][0].value`

does that make it a little clearer?

Comment (ZenVega) [reply app/src/plotter.py:348]: One more info. Before, we used panels CheckBoxGroup-Widget to render the multi_choice_filters. In order to transform it to a legend like this: 
image
we need access every Row in this "Column" : 
image
This is the cause of that additional nesting

File: app/src/plotter.py (hunk 8 above)
Comment (guadiromero) [reply app/src/plotter.py:327]: Ah, got it 👍

File: app/src/layout.py (hunk 1 above)
Comment (guadiromero) [reply app/src/layout.py]: You are right! But it can still be simplified a bit, I commited a couple small changes.

- As the function inside `create_update_chart` is only used once inside of it, we can make use python's lambda (an anonymous function with a simpler one-line syntax)
- This function should be defined outside the loop, otherwise we define it over and over with each loop, which is unnecessary

**Comment by guadiromero** [comment]
@ZenVega Looks good now :)

Make sure to:
- Pull the latest changes from main and fix the git conflicts
- Run the new pre-commit checks introduced by Matthieu to make sure the style is correct

Then you should be ready for merge 🚀

File: app/plot/plotter.py [hunk 12]
```diff
@@ -442,9 +527,16 @@ def create_plot(self):
             plot.width_policy = "max"
 
             # Add a label in the center
+            highlight_category = self.config["filters_defaults"][
+                "single_choice_highlight"
+            ]
```
Comment (MRigal) [inline app/plot/plotter.py:532]: This was already defined before and is not depending on the code of the config. Or there is something missing there...

File: app/plot/plotter.py [hunk 13]
```diff
@@ -442,9 +527,16 @@ def create_plot(self):
...
             for key, value, color in zip(
-                self.raw_data[code][single_choice_key]["x"],
-                self.raw_data[code][single_choice_key]["y"],
+                self.raw_data[code][self.config["filters_defaults"]["single_choice"]][
+                    "x"
+                ],
+                self.raw_data[code][self.config["filters_defaults"]["single_choice"]][
+                    "y"
+                ],
```
Comment (MRigal) [inline app/plot/plotter.py:539]: I don't know why this change was made, this was exactly working before and making the graph more readable. Also here we are re-reading/parsing a dict inside a double loop, instead of reading it once and reusing it afterwards

File: app/styles/main.css (hunk 11 above)
Comment (MRigal) [reply app/styles/main.css:64]: I guess too late... :-/

**Review by MRigal — COMMENTED** [review]
@ZenVega maybe you can give a look at the comments I've left when you have time :-)
